jekyll/_cci2/github-integration.adoc

```diff
@@ -0,0 +1,279 @@
+
+Add a `.circleci` directory to a repository you want to connect to CircleCI. Inside that directory, add a `config.yml` file. You can use the basic configuration below inside your configuration file:
+
+```yaml
+version: 2.1
+
+orbs:
+    hello: circleci/hello-build@0.0.5
+
+workflows:
+    "Hello Workflow":
+        jobs:
+          - hello/hello-build
+
+```
+
+After you create and commit a `.circleci/config.yml` file to your GitHub repository, CircleCI immediately checks your code out and runs your first job along with any configured tests. This basic example runs a "hello world."
+
+CircleCI runs your tests on a clean container every time so that your tests are fresh each time you push code, and so that your code is never accessible to other users. Watch your tests update in real-time on https://circleci.com/dashboard[your dashboard]. You can also get status updates through email notifications, or look for the status badges that appear on GitHub. Integrated statuses also appear on the pull request screen, to show that all tests have passed.
+
+== Enable your project to check out additional private repositories
+
+If your testing process refers to multiple repositories, CircleCI will need a GitHub user key in addition to the deploy key because each deploy key is valid for only _one_ repository while a GitHub user key has access to _all_ of your GitHub repositories.
+
+Provide CircleCI with a GitHub user key in your project's **Project Settings** > **SSH keys**. Scroll down the page to **User Key** and click **Authorize with Github**. CircleCI creates and associates this new SSH key with your GitHub user account for access to all your repositories.
+
+== Best practices for keys
+
+- Use Deploy Keys whenever possible.
+- When Deploy Keys cannot be used, <<#controlling-access-via-a-machine-user,Machine User Keys>> must be used, and have their access restricted to the most limited set of repos and permissions necessary.
+- Never use non-Machine user keys (keys should be associated with the build, not with a specific person).
+- You must rotate the Deploy or User key as part of revoking user access to that repo.
+  1. After revoking the user’s access in GitHub, delete keys in GitHub.
+  1. Delete the keys in the CircleCI project.
+  1. Regenerate the keys in CircleCI project.
+- Ensure no developer has access to a build in a repo with a User Key that requires more access than they have.
+
+== Establish the authenticity of an SSH host
+
+When using SSH keys to checkout repositories, it may be necessary to add the fingerprints for GitHub to a "known hosts" file (`~/.ssh/known_hosts`) so that the executor can verify that the host it is connecting to is authentic. The `checkout` job step does this automatically, so you will need to run the following commands if you opt to use a custom checkout command:
```

Comment:
```suggestion
When using SSH keys to check out repositories, it may be necessary to add the fingerprints for GitHub to a "known hosts" file (`~/.ssh/known_hosts`) so that the executor can verify that the host it is connecting to is authentic. The `checkout` job step does this automatically, so you will need to run the following commands if you opt to use a custom checkout command:
```